packages/core/react/src/WalletProvider.tsx

```diff
@@ -116,10 +116,9 @@ export const WalletProvider: FC<WalletProviderProps> = ({
     const select = useCallback(
         async (walletName: WalletName | null) => {
             if (name === walletName) return;
-            if (adapter) await adapter.disconnect();
             setName(walletName);
         },
-        [name, adapter, setName]
```

Comment:
Removed `setName` from the dependencies array. See https://github.com/solana-labs/wallet-adapter/pull/216.